Failing test: Jest Tests.x-pack/plugins/security_solution/public/detection_engine/rule_gaps/components/manual_rule_run - ManualRuleRunModal should render confirmation button disabled if selected end date is in future (#186189)

e40pud opened the pull request:
## Summary

Related tickets https://github.com/elastic/kibana/issues/186189 and https://github.com/elastic/kibana/issues/186187, and https://github.com/elastic/kibana/issues/186188

Attempt to fix failing manual rule run tests added in this PR https://github.com/elastic/kibana/pull/184500

e40pud committed:
Failing test: Jest Tests.x-pack/plugins/security_solution/public/detection_engine/rule_gaps/components/manual_rule_run - ManualRuleRunModal should render confirmation button disabled if selected end date is in future (#186189)

elasticmachine commented:
Pinging @elastic/security-solution (Team: SecuritySolution)

elasticmachine commented:
Pinging @elastic/security-detection-engine (Team:Detection Engine)

e40pud commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into security/tests/186189-failing-rule-gaps

rylnd commented on x-pack/plugins/security_solution/public/detection_engine/rule_gaps/components/manual_rule_run/index.test.tsx:
```diff
@@ -6,73 +6,78 @@
-    const wrapper = render(
-      <ManualRuleRunModal onCancel={onCancelMock} onConfirm={onConfirmMock} />
-    );
+    render(<ManualRuleRunModal onCancel={onCancelMock} onConfirm={onConfirmMock} />);
```
Are these subsequent `render` calls necessary?

If the elements are assigned in the `beforeEach`, subsequent `render`s aren't going to update those references, right?

e40pud replied:
No these renders should be removed! I'm not sure why those references still work though

rylnd reviewed (approved):
@e40pud these changes look reasonable, but I'm not seeing any explanation as to what the problem was (that caused the NoMethodError for `scrollIntoView`), or how these changes address that problem.

If you could provide some of that context, that would help both review and posterity. Thanks!

e40pud commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into security/tests/186189-failing-rule-gaps

e40pud committed:
Review feedback

e40pud committed:
Attempt to fix the `scrollIntoView` is not a function error

e40pud commented:
> @e40pud these changes look reasonable, but I'm not seeing any explanation as to what the problem was (that caused the NoMethodError for `scrollIntoView`), or how these changes address that problem.
> 
> If you could provide some of that context, that would help both review and posterity. Thanks!

I was never able to reproduce the issue locally and thus decided to restructure tests and used methods. My initial thoughts were that the cause is the `getByText` method which calls `scrollIntoView` method for whatever reasons.

Now, I found this discussion https://stackoverflow.com/a/53294906 and the link to the issue https://github.com/jsdom/jsdom/issues/1695. Updated tests and added proposed solution.

rylnd commented:
That's great info, thanks @e40pud!
